src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,50 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't create phone " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
-    public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+    public List<Phone> findAll(Map<String, String[]> parameters) {
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            List<Predicate> andPredicateList = new ArrayList<>();
+            for (Map.Entry<String, String[]> entry : parameters.entrySet()) {
+                String[] stringArray = entry.getValue();
+                List<Predicate> orPredicateList = new ArrayList<>();
+                for (String parameter : stringArray) {
```

Comment:
let's use `in()` method on `phoneRoot.get(entry.getKey()` instead

Reply:
Ok. ^__^